src/app/forms/sensor/sensor.component.ts

```diff
@@ -0,0 +1,310 @@
+        if (longitudeUpdated || latitudeUpdated || heightUpdated) {
+          try {
+            await this.sensorService.updateLocation(this.sensor._id, updatedSensorProperties.location);
+          } catch (error) {
+            console.error(error);
+          }
+        }
+
+        await this.router.navigate(['']);
+      } else {
+        const dataStreams = [];
+        const formDataStreams = this.form.value.dataStreams || [];
+        for (const dataStream of formDataStreams) {
+          dataStreams.push({
+            name: dataStream.name,
+            reason: dataStream.reason || undefined,
+            description: dataStream.description || undefined,
+            observedProperty: dataStream.observedProperty || undefined,
+            isPublic: dataStream.isPublic,
+            isOpenData: dataStream.isOpenData,
+            isReusable: dataStream.isReusable,
+            documentationUrl: dataStream.documentationUrl || undefined,
+            dataLink: dataStream.dataLink || undefined,
+            unitOfMeasurement: dataStream.unitOfMeasurement || undefined,
+            dataFrequency: dataStream.dataFrequency,
+            dataQuality: dataStream.dataQuality,
+          });
+        }
+
+        const sensor: IRegisterSensorBody = {
+          dataStreams,
+          typeName: this.form.value.type.typeName,
+          location: this.form.value.location || {},
+          active: JSON.parse(this.form.value.active.value.toLowerCase()) || false, // cast strings ("true") to boolean
+          aim: this.form.value.aim,
+          description: this.form.value.description,
+          documentationUrl: this.form.value.documentationUrl || undefined,
+          manufacturer: this.form.value.manufacturer || undefined,
+          name: this.form.value.name,
+          theme: this.form.value.theme ? this.form.value.theme.value : [],
+          typeDetails: {subType: this.form.value.type.typeDetails || ''},
+        };
+
+        try {
+          await this.sensorService.register(sensor);
+
+          this.locationService.showLocation(null);
+          this.alertService.success('Sensor registered!', false, 4000);
+        } catch (error) {
+          this.alertService.error(`An error has occurred while creating sensor: ${error}.`);
+        }
+      }
+    } else {
+      this.alertService.error(`The form is invalid.`);
+    }
+  }
+
+  public ngOnInit() {
+    const urlRegex = '(https?://)?([\\da-z.-]+)\\.([a-z.]{2,6})[/\\w .-]*/?';
+
+    this.form = this.formBuilder.group({
+      name: ['', [Validators.required, Validators.minLength(6)]],
+      aim: '',
+      description: '',
+      manufacturer: '',
+      active: '',
+      documentationUrl: ['', [Validators.pattern(urlRegex)]],
+      dataStreams: new FormArray([]),
+      location: [],
+      type: [],
+      theme: [],
+    });
+
+    this.formControlSteps = {
+      'Sensor Properties': [
```

Comment:
Tof hoe de verschillende form controls zo over de pagina's heen gelegd kunnen worden. 

Kun je de labels in de HTML zetten? Zo gaan we ze niet kunnen vertalen

EDIT: we hebben op meer plekken dat we text tonen vanuit de TS, zoals de Alert box. We zullen moeten uitzoeken of we die ook i18n proof kunnen maken